### ISSUE TITLE
2020 call for push bit/contributors for releases and issue and general contrib work 

### ISSUE THREAD

**Issue by cartazio** [haskell/vector#267]
I've been remiss in how i've handled cutting releases, and everything is better with some collaboratorators (i definitely prioritize better with some good collab.). I was also a bit preoccupied with making sure some changes to primitive in 2018/2019 moved along in a good way as @chessai and @andrewthad can recall/attest

@Boarders @gwils, @chessai @andrewthad  @RyanGlScott  et al: 
let whoemever wants to be more active to contrib and release prep communicate thusly on this ticket or via email and let loose the changelogs of war.

active email/irc/chat app is the best communication mechanism for collab or release engineering things aside from the pure concrete procedural stuff for me (i'm a tad high strung at times, though i mostly manage it ok). I like to keep tickets concrete and actionable or about articulating technical questions :)

**Comment by Boarders** [comment]
@lehins

**Comment by cartazio** [comment]
or just getting push perms plus yelling at me to cut a release every once in a while

**Comment by lehins** [comment]
@Boarders, I thought you'd never ask ;) Sure, I would be happy to become one of `vector` maintainers.

I do have one condition though, unlike @cartazio, I prefer to keep communication for open source projects open to the community as well, unless, of course, it is a sensitive discussion of some security issue. Therefore I will not be using private chats or emails for communication, except in rare cases when information discussed really needs to stay private. Naturally, extensive discussions of organizational problems have no place in github issues, so a public dedicated `vector` chat room is certainly a better place for such purpose. Not sure if a room like this already exists somewhere, but we can definitely create one. Gitter comes to mind, but I have no preference.

> or just getting push perms plus yelling at me to cut a release every once in a while

@cartazio, that would be fine. I am not that eager to handle such important responsibility as cutting actual releases.

**Comment by Boarders** [comment]
@lehins: I agree with you it is better for all sorts of reasons for these discussions to be open. For one selfish one, I have learnt a huge amount from the issues of various projects and it would be a shame for all of that knowledge to merely go to waste in private discussions. I think more importantly it allows for a greater accountability to the community and allows anyone (and not just those with back channel access) to these discussions. Particularly if the discussions are those of a technical nature (including how to organise the project) I would have a strong preference for them to be out in public.

**Comment by cartazio** [comment]
@lehins on freenode #numerical-haskell i'd suggest :)

**Comment by cartazio** [comment]
agreed that design design should be explained.

**Comment by lehins** [comment]
> @lehins on freenode #numerical-haskell i'd suggest :)

@cartazio I am not to keen on the idea of using IRC for this. Requiring people to install a special client to view/participate in a public discussion is a bit dinosaur ;) Aferall, it's 2020, surely we can use a browser for communicating with each other. 

Ideally it would be a public channel, which allows linking directly to any particular discussion from github issues. Most importantly though, anyone should be able to read discussions without being required to install anything or register with the service. Not sure if there is a service that gives IRC such capabilities, if there is, then fine, we can provide instructions in the readme on how to use it.

I'd also suggest a channel that is dedicated to `vector`, not to a broad topic of numerical computing in Haskell. Arguments are very simple:
* There are already a few places that I know of that discuss numerical computing in Haskell: data-haskell, accelerate, numerical-haskell, massiv, etc. None of them are a dependency of vector. More over there is no point in favoring one community over another.
* There are plenty of users of `vector` who simply don't care about numerical processing.

**Comment by cartazio** [comment]
@lehins  its a public channel, its got ~ 30 fifty people who are into high performance haskell and mathematical computing

**Comment by cartazio** [comment]
most of the nontrivial support questions i see about vector really get into when does fusion work well vs when does it fail, and how ghc optimizes various flavors of code

**Comment by cartazio** [comment]
hrm, write perms will have to wait, i'll circle back on it after i catchup on outstanding prs etc.
Looks like it requires coordinating with folks with admin perms which i'll look into after i review all the prs

**Comment by cartazio** [comment]
I’ll think about all of this more next week after I cool down (literally,
mild fever this week ;) )

You make good points I just need to think about em.

**Comment by merijn** [comment]
I've got some stuff on my todo list to push into vector if I ever get around to it ;)

**Comment by gwils** [comment]
There is a web chat facility on freenode that works with just a browser, but I only really recommend it for quick drop-ins rather than long-term connections.
http://webchat.freenode.net/?randomnick=1&channels=%23numerical-haskell&uio=d4

A feature of IRC that might be a problem here is its ephemeral nature - anyone in the channel can see what's being said, but anyone in the future will not be able to access those past conversations. Something like gitter maintains history.

Personally I use IRCCloud to maintain a constant IRC connection available from all my devices, but I pay for that service. ZNC is a free solution here, but it requires setting up and running a server.

**Comment by fumieval** [comment]
@cartazio seems to dislike GitHub issues to the point that the one I posted got renamed & locked (#266), but I'm not sure if IRC is any better. Since IRC messages are private and ephemeral, people would end up explaining the same thing to different people.

**Comment by cartazio** [comment]
.  I’ve told you how to reach out.  The rest is on you.

Tickets are for concrete problem solving and user confusion/education.  I’m
all for education on the tickets.  For everthing else use other
communication tools. End of story.

**Comment by Boarders** [comment]
Limiting what issues can be brought up on github (the public facing part of the library) with a library under the purview of the CLC just seems flat out wrong to me (as does re-naming issues that you personally do not like the tone or style of). I seriously think some reform needs to take place if this is the best on offer because it is not good enough. The vector library is for the whole haskell community and we should be able to put issues (of all the usual sorts) on github. I don't know why this needs saying!

**Comment by cartazio** [comment]
I’m trying to constructively address communication problems in a way that
works for me. Your failure to be willing to acknowledge that different
people communicate best in different tools and fashions is at issue.

I have generalized anxiety disorder that I actively work to manage and
y’all are being thick headed assholes where I need to spell that out.  Stop
it.  Please.

Or go away.  Please.

Different people are comfortable discussing complicated matters through
different tools.  And you should respect that other humans are different
and when they Express reasonable communication preferences you really
shouldn’t moralize that preferences that aren’t yours are invalid.

**Comment by fumieval** [comment]
In the issue locked by you people were discussing how to improve the release process. I think that's useful information for all users. I mean I can explain about this to you privately, but I'd (and probably you would) end up in writing the same thing somewhere else so that other users can understand it. I'm not sure that's what really want though. Am I still being an arsehole?

**Comment by cartazio** [comment]
For matters of interpersonal communication just email me or let’s do a
video chat. Tickets always read as adversarial in my head when they veer
outside technical matters.  Please respect this.  I’ve tried to communicate
this quite explicitly.

**Comment by Boarders** [comment]
It is perfectly fine if you don't want to engage on github issues and work better via irc, but we need a maintainer who is willing to allow and engage in discussion on github issues and who you can then talk to via other means if you so wish. I would appreciate CLC input here @gwils, I don't wish to be adversarial but I don't know how else to express this.

**Comment by cartazio** [comment]
@Boarders i added 4 people to have contrib perms. They can chime in if they want but that’s their personal choice. Privacy is a right.

**Comment by Boarders** [comment]
Sounds good, so long as you leave the issues you don't want to deal with or find unpleasant for them (and they are left open) then that works!
